cuebot/src/main/java/com/imageworks/spcue/service/HostManagerService.java

```diff
@@ -152,7 +152,28 @@ public DispatchHost createHost(HostReport report) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public DispatchHost createHost(RenderHost rhost) {
-        return createHost(rhost, getDefaultAllocationDetail());
+        // Find suitable allocation with facility and tags.
+        AllocationEntity alloc = null;
+        if (rhost.getTagsCount() > 0) {
+            String facility = rhost.getFacility();
+            for (String tag : rhost.getTagsList()) {
+                try {
+                    alloc = allocationDao.findAllocationEntity(facility, tag);
+                    logger.info("set " + rhost.getName() +
+                            " to the given allocation " + alloc.getName());
+                    break;
+                }
+                catch (Exception e) {
```

Comment:
Is there a more specific exception we can catch here?

Reply:
Yes, updated with EmptyResultDataAccessException.
Also updated unit test for the default allocation.